### PR TITLE
Implement debug and display trait for errors that can generate panics

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Reason for This PR

Implement Debug & Display for Errors that can generate panics #856

## Description of Changes

I've added mostly Display traits to describe friendlier user-facing errors from api_server, which inherits errors from vmm, devices, arch etc.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided. 
- [x] The description of changes is clear and encompassing.
- [x] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [x] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [x] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.

### CODE REVIEW

File: net_util/src/tap.rs [hunk 1]
```diff
@@ -31,6 +32,17 @@ pub enum Error {
 
 pub type Result<T> = ::std::result::Result<T, Error>;
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Error::OpenTun(err) => write!(f, "Couldn't open /dev/net/tun: {}", err),
+            Error::CreateTap(err) => write!(f, "Unable to create tap interface: {}", err),
+            Error::IoctlError(err) => write!(f, "Ioctl failed: {}", err),
+            Error::NetUtil(err) => write!(f, "Failed to create a socket: {}", err),
```
[the remark this answers is not in the document]
Comment (iulianbarbu) [reply net_util/src/tap.rs]: Updated.

File: net_util/src/tap.rs [hunk 2]
```diff
@@ -31,6 +32,17 @@ pub enum Error {
 
 pub type Result<T> = ::std::result::Result<T, Error>;
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Error::OpenTun(err) => write!(f, "Couldn't open /dev/net/tun: {}", err),
+            Error::CreateTap(err) => write!(f, "Unable to create tap interface: {}", err),
+            Error::IoctlError(err) => write!(f, "Ioctl failed: {}", err),
```
[the remark this answers is not in the document]
Comment (iulianbarbu) [reply net_util/src/tap.rs]: Updated.

File: memory_model/src/guest_memory.rs [hunk 3]
```diff
@@ -35,6 +36,40 @@ pub enum Error {
 }
 type Result<T> = result::Result<T, Error>;
 
+impl fmt::Display for Error {
```
[the remark this answers is not in the document]
Comment (iulianbarbu) [reply memory_model/src/guest_memory.rs]: It is not necessary. 
Updated accordingly.

File: memory_model/src/guest_memory.rs [hunk 4]
```diff
@@ -35,6 +36,40 @@ pub enum Error {
...
+                err
+            ),
+            Error::MemoryMappingFailed(err) => write!(
+                f,
+                "Failure in creating an anonymous shared mapping: {}",
+                err
+            ),
+            Error::MemoryNotInitialized => write!(f, "Failure in initializing guest memory."),
+            Error::MemoryRegionOverlap => write!(f, "Two of the memory regions are overlapping."),
+            Error::NoMemoryRegions => write!(
```
[the remark this answers is not in the document]
Comment (iulianbarbu) [reply memory_model/src/guest_memory.rs]: Why do you think this error should be replaced with an expect?

File: arch/src/x86_64/mod.rs [hunk 5]
```diff
@@ -39,6 +40,19 @@ pub enum Error {
...
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Error::E820Configuration => write!(f, "Invalid e820 setup params."),
+            Error::MpTableSetup(err) => write!(f, "Error writing MP table to memory: {}", err),
+            Error::ZeroPagePastRamEnd => {
+                write!(f, "The zero page extends past the end of guest_mem.")
+            }
+            Error::ZeroPageSetup => write!(f, "Error writing the zero page of guest memory."),
```
Comment (dianpopa) [inline arch/src/x86_64/mod.rs]: It Should be `Error writing the zero page to guest memory.` The comment from aboce the correspondent enum item should be updated too.

**Comment by dianpopa** [comment]
Also related to #593.

**Comment by iulianbarbu** [comment]
Close this PR since it needs to be divided in two to be easy reviewed.

**Pull request closed without merging**